plugins/woocommerce/src/Admin/API/OnboardingPlugins.php

```diff
@@ -183,6 +210,39 @@ function( $action ) use ( $job_id ) {
 		return $response;
 	}
 
+
+	/**
+	 * Return Jetpack authorization URL.
+	 *
+	 * @param WP_REST_Request $request WP_REST_Request object.
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	public function get_jetpack_authorization_url( WP_REST_Request $request ) {
+		$manager = new Manager( 'woocommerce' );
+		// Register the site to wp.com.
+		if ( ! $manager->is_connected() ) {
+			$result = $manager->try_registration();
+			if ( is_wp_error( $result ) ) {
+				 throw new \Exception( $result->get_error_message() );
+			}
+		}
+
+		$redirect_url = $request->get_param( 'redirect_url' );
+		$calypso_env  = defined( 'WOOCOMMERCE_CALYPSO_ENVIRONMENT' ) && in_array( WOOCOMMERCE_CALYPSO_ENVIRONMENT, [ 'development', 'wpcalypso', 'horizon', 'stage' ], true ) ? WOOCOMMERCE_CALYPSO_ENVIRONMENT : 'production';
+
+		return [
+			'url' => add_query_arg(
+				[
+					'from'        => $request->get_param( 'from' ),
```

Comment:
I think we still need to set a default to `woocommerce-onboarding` or `woocommerce-core-profiler` here

Reply:
The default is set in [here](https://github.com/woocommerce/woocommerce/pull/38674/files#diff-ccca16b15b2dfeb741b98e5d2695dbf1d789de8a44fb869e426929f62f681863R119)

Reply:
I missed that, thanks!